File: ruby/v2.rb
#!/usr/bin/env ruby

require 'benchmark'

def max_subarray(v,size)
    best = -1.0/0.0
    i = 0
    while i < size do
        j = i
        sum = 0
        while j < size do
			sum = sum + v[j]
            if sum > best
                best = sum
			end
        	j+=1
		end
    	i+=1
	end
    if best > 0
        return best
	end
    return 0
end
        
if ARGV.length < 2
	print("Argumentos invalidos")
	exit;
end
size = ARGV[0].to_i
f = open(ARGV[1],"r")
v = []
while size > 0 do
	v.push(f.readline().to_i)
    size = size - 1
end
size = ARGV[0].to_i
result = 0;
time = Benchmark.measure{
    result = max_subarray(v,size)
}
if ARGV.length > 2
    if ARGV[2] == "-t"
        puts(time.real)
    elsif ARGV[2] == "-r"
        puts(result)
    end
end

File: python/v2.py
def max_subarray(v,size):
    best = -math.inf
    i = 0
    while i < size:
        j = i
        sum = 0
        while j < size:
            sum = sum + v[j]
            if sum > best:
                best = sum
            j+=1
        i+=1
    if math.isfinite(best):
        return best
    return 0
        
if __name__ == '__main__':
    import math
    import sys
    import time
    if len(sys.argv) < 3:
        print("Argumentos invalidos")
        sys.exit(0) 
    size = int(sys.argv[1])
    f = open(sys.argv[2],"r")
    v = []
    while size > 0:
        v.append(int(f.readline()))
        size = size - 1
    size = int(sys.argv[1])
    start = time.clock()
    result = max_subarray(v,size)
    end = time.clock()
    if len(sys.argv) > 3 and sys.argv[3] == "-t":
        print(end - start)
    if len(sys.argv) > 3 and sys.argv[3] == "-r":
        print(result)
